adam/ontology/ontology.py

```diff
@@ -122,22 +122,28 @@ def nodes_with_properties(
         required_properties: Iterable["OntologyNode"] = immutableset(),
         *,
         banned_properties: AbstractSet["OntologyNode"] = immutableset(),
+        banned_ontology_types: AbstractSet["OntologyNode"] = immutableset(),
```

Comment:
@spigo900 : Please add a unit test for this.

Reply:
Done.